examples/some-verifier/src/main.rs

```diff
@@ -287,213 +300,227 @@ async fn add_verification(
-        return Err(Error::InvalidTimestamp);
-    }
-
-    let bi = state
-        .node_client
-        .get_block_info(BlockIdentifier::LastFinal)
+    match state
+        .database
+        .remove_verification(credential, platform)
         .await
-        .map_err(|e| Error::CredentialLookup(e.into()))?;
-    let public_data =
-        web3id::get_public_data(&mut state.node_client, state.network, &proof, bi.block_hash)
-            .await?;
-
-    // Check that all credentials are active at the time of the query
-    if !public_data
-        .iter()
-        .all(|cm| matches!(cm.status, CredentialStatus::Active))
     {
-        return Err(Error::InactiveCredentials);
-    }
-    // And then verify the cryptographic proofs
-    let request = proof.verify(
-        &state.crypto_params,
-        public_data.iter().map(|cm| &cm.inputs),
-    )?;
-
-    // Check that the challenge is the hash of the supplied timestamp
-    let iso_time = timestamp.to_rfc3339_opts(SecondsFormat::Millis, true);
-    let mut hasher = Sha256::new();
-    hasher.update(iso_time.as_bytes());
-    let hash = hasher.finalize();
-
-    if &request.challenge[..] != &hash[..] {
-        return Err(Error::InvalidChallenge);
+        Ok(removed) => {
+            if removed {
+                tracing::debug!("Successfully removed verification for {credential}.");
+                Ok(StatusCode::OK)
+            } else {
+                tracing::debug!(
+                    "Request to remove a verification with a non-existing credential id."
+                );
+                Ok(StatusCode::BAD_REQUEST)
+            }
+        }
+        Err(err) => {
+            tracing::warn!("Error removing entries for {credential}: {err}");
+            Err(Error::Database(err))
+        }
     }
-
-    Ok(())
 }
 
-fn proof_to_verifications_entry(
-    state: AppState,
-    proof: &CredentialProof<ArCurve, Web3IdAttribute>,
-    entry: &mut VerificationsEntry,
-) -> Result<(), Error> {
-    match proof {
-        // Platform verification (Telegram, Discord, etc.)
-        CredentialProof::Web3Id {
-            network,
-            contract,
-            proofs,
-            holder,
-            ..
-        } => {
-            if network != &state.network {
-                return Err(Error::InvalidIssuer);
-            }
+/// Verify the request. In particular this checks
+/// - all credentials mentioned in the request exist, and are active (in
+///   particular they have not expired)
+/// - all credentials are on the required network
+/// - cryptographic proofs are valid
+/// - the timestamp in the request is no more than 10min from present.
+impl AppState {
+    async fn verify_request(
+        &mut self,
+        request: &Request,
+    ) -> Result<web3id::Request<ArCurve, Web3IdAttribute>, Error> {
+        let Request { proof, timestamp } = request;
+
+        let delta = Utc::now().signed_duration_since(*timestamp);
+        if delta.num_minutes().abs() > 10 {
```

Comment:
Just a note on the time: The "timer" starts when the proof is requested from the wallet, so if the user takes a long time before accepting, the request will get rejected.

Reply:
Yes. But if they do then they can retry. 24h is too much time.

The best case would be if the frontend handled this and just told the user immediately that they need to retry if they take too long. But this is not really relevant for the MVP.